### PR TITLE
ROC-4520 reject defendant company repayment plan and update dashboard

### PR DESCRIPTION
**Pull request by rattur**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-4520

### Change description

Claimant response rejects defendant company repayment plan

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [ x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x ] No

### CODE REVIEW

File: src/main/features/dashboard/views/macro/caseStatus.njk [hunk 1]
```diff
@@ -67,106 +67,118 @@
 {% macro caseStatusForClaimant(status, claim) %}
   {% if status === ClaimStatus.NO_RESPONSE %}
     {{ noResponseForClaimantDashboard() }}
-  {% elif status === ClaimStatus.MORE_TIME_REQUESTED %}
+    {% elif status === ClaimStatus.MORE_TIME_REQUESTED %}
```
Comment (andrewwa-kainos) [inline src/main/features/dashboard/views/macro/caseStatus.njk]: Try to avoid unnecessary changes - they pollute the PR. Ideally I'd like to see these indents reverted. (Seems that the auto-formatter doesn't understand `elif`.)

File: src/main/features/dashboard/views/macro/caseStatus.njk [hunk 2]
```diff
@@ -67,106 +67,118 @@
...
-          {{ claimPartAdmittedToOweForDefendantDashboard(claim) }}
-        {% endif %}
-      {% elif claim.response.responseType === domain.ResponseType.FULL_ADMISSION %}
-        {{ claimFullAdmissionForDefendantDashboard(claim) }}
+      {% if claim.response.paymentDeclaration %}
+        {{ claimPartAdmittedAlreadyPaidForDefendantDashboard(claim) }}
+      {% else %}
+        {{ claimPartAdmittedToOweForDefendantDashboard(claim) }}
       {% endif %}
+      {% elif claim.response.responseType === domain.ResponseType.FULL_ADMISSION %}
```
Comment (andrewwa-kainos) [inline src/main/features/dashboard/views/macro/caseStatus.njk]: I'm not sure about
```
{% endif %}
{% elif ... %}
```
Can you double-check these `elif` branches can be accessed?

**Comment by rattur** [comment]
Andrew yes we need to chat about this story https://tools.hmcts.net/jira/browse/ROC-4519

File: src/main/features/dashboard/views/macro/caseStatus.njk (hunk 1 above)
Comment (rattur) [reply src/main/features/dashboard/views/macro/caseStatus.njk]: True, I think its the auto formatter

File: src/main/features/dashboard/views/macro/caseStatus.njk (hunk 2 above)
Comment (rattur) [reply src/main/features/dashboard/views/macro/caseStatus.njk]: Yes they are accessed

File: src/main/features/claimant-response/views/confirmation/proposed-alternate-payment-plan-to-business-defendant.njk [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<h2 class="heading-medium">Send financial details</h2>
```
Comment (robertparkinson) [inline src/main/features/claimant-response/views/confirmation/proposed-alternate-payment-plan-to-business-defendant.njk]: missing translation

File: src/main/features/claimant-response/views/confirmation/proposed-alternate-payment-plan-to-business-defendant.njk [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+<p>{{ t('Make sure you include your claim number: {{ claimNumber }}.', { claimNumber: claim.claimNumber }) }}</p>
+
+<p>{{ t('The court will review the details to see if the defendant can afford your repayment plan.') }}</p>
+
+<h2 class="heading-medium">{{ t('Use this address') }}</h2>
+
+<p>{{ t('The defendant should have sent you the details - send them to:') }}</p>
+
+<p>
```
Comment (robertparkinson) [inline src/main/features/claimant-response/views/confirmation/proposed-alternate-payment-plan-to-business-defendant.njk]: use a div with the `keep-formatting panel` classes as can be found in the send-your-response-by-email.njk

File: src/main/features/dashboard/views/macro/claimStatus/claimFullAdmissionForDefendant.njk [hunk 5]
```diff
@@ -5,14 +5,25 @@
...
 {% macro claimFullAdmissionForDefendantDashboard(claim) %}
 
-  {% if claim.response.paymentIntention.paymentOption === domain.PaymentOption.IMMEDIATELY %}
-    {{ t('You’ve admitted all of the claim and said you’ll pay the full amount immediately.') }}
-  {% elseif claim.response.paymentIntention.paymentOption === domain.PaymentOption.BY_SPECIFIED_DATE %}
-    {{ t('You’ve admitted all of the claim and offered to pay the full amount by {{ paymentDate }}.',
-      { paymentDate: claim.response.paymentIntention.paymentDate | date }) }}
-  {% elseif  claim.response.paymentIntention.paymentOption === domain.PaymentOption.INSTALMENTS %}
-    {{ t('You’ve admitted all of the claim and offered to pay the full amount in instalments.') }}
+  {% if claim.claimData.defendant.isBusiness() %}
```
Comment (robertparkinson) [inline src/main/features/dashboard/views/macro/claimStatus/claimFullAdmissionForDefendant.njk]: Please check with content guys as this content is almost identical. Ask them to update the prototype so both versions are the same then remove the if statement for isBusiness.

File: src/main/features/response/views/confirmation/full-admission.njk [hunk 6]
```diff
@@ -11,9 +12,15 @@
   <p>{{ t('We’ve emailed {{ claimantName }} your offer to pay by {{ paymentDate }} and your explanation of why you can’t pay before then.',
```
Comment (robertparkinson) [inline src/main/features/response/views/confirmation/full-admission.njk]: Please update all these if endif statements to be if elif endif for the payment options.

File: src/main/features/response/views/send-company-financial-details.njk [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+{% block content %}
+  <div class="grid-row">
+    <div class="column-two-thirds">
+      <form novalidate method="post">
+        {{ csrfProtection(csrf) }}
+        <p>{{ t('You need to send {{ claimantName }} your company or organisation’s most recent statement of accounts.', { claimantName: claimantFullName }) }}</p>
+        <p>{{ t('They’ll review your accounts and can reject your suggested repayment plan if they believe you can pay sooner.') }}</p>
+        <p>{{ t('If they reject your plan, the court will make a new plan based on your financial details.') }}</p>
+        {{ submitButton('Continue') }}
```
Comment (robertparkinson) [inline src/main/features/response/views/send-company-financial-details.njk]: this doesn't need to be a form. can just be an internalLink with the button css to the next page.

File: src/main/app/claims/claimStoreClient.ts [hunk 8]
```diff
@@ -235,7 +235,8 @@ export class ClaimStoreClient {
   }
 
   saveClaimantResponse (claim: Claim, draft: Draft<DraftClaimantResponse>, user: User): Promise<void> {
```
Comment (markry-kainos) [inline src/main/app/claims/claimStoreClient.ts]: @robertparkinson @varma02 @rattur pls review - these changes are to fix claimant response with refer to judge formalisation option (1 of 2)

File: src/main/app/claims/converters/claimantResponseConverter.ts [hunk 9]
```diff
@@ -14,7 +14,10 @@ import { MomentFactory } from 'shared/momentFactory'
 
 export class ClaimantResponseConverter {
 
-  public static convertToClaimantResponse (draftClaimantResponse: DraftClaimantResponse): ClaimantResponse {
```
Comment (markry-kainos) [inline src/main/app/claims/converters/claimantResponseConverter.ts]: @robertparkinson @varma02 @rattur pls review - these changes are to fix claimant response with refer to judge formalisation option (2 of 2)

File: src/main/app/claims/models/claim.ts [hunk 10]
```diff
@@ -240,6 +241,10 @@ export class Claim {
     return statuses
   }
 
+  private isDefendantBusiness (): boolean {
```
[the remark this answers is not in the document]
Comment (rattur) [reply src/main/app/claims/models/claim.ts]: I think its a cleaner solution just to call the function

File: src/main/app/claims/claimStoreClient.ts (hunk 8 above)
Comment (rattur) [reply src/main/app/claims/claimStoreClient.ts]: This looks good to me by passing isDefendantBusiness within the ClaimStoreClient and then checking in the ClaimantResponseConverter for isBusiness and setting REFER_TO_JUDGE

File: src/main/features/response/routes/check-and-send.ts [hunk 11]
```diff
@@ -43,8 +43,8 @@ function defendantIsCounterClaiming (draft: Draft<ResponseDraft>): boolean {
 }
 
 function isStatementOfTruthRequired (draft: Draft<ResponseDraft>): boolean {
-  const responseType: ResponseType = draft.document.response.type
-  return (responseType === ResponseType.DEFENCE && !defendantIsCounterClaiming(draft))
+  // const responseType: ResponseType = draft.document.response.type
```
[the remark this answers is not in the document]
Comment (singh-surjeet) [reply src/main/features/response/routes/check-and-send.ts]: @rattur is that resolved?

File: src/main/app/claims/models/claimant-response/court-determination/decisionType.ts [hunk 12]
```diff
@@ -2,5 +2,6 @@ export enum DecisionType {
   CLAIMANT_IN_FAVOUR_OF_DEFENDANT = 'CLAIMANT_IN_FAVOUR_OF_DEFENDANT',
```
Comment (robertparkinson) [inline src/main/app/claims/models/claimant-response/court-determination/decisionType.ts]: we now have two of these. Please remove one and update the references to the other one.

File: src/main/features/claimant-response/draft/courtDecision.ts [hunk 13]
```diff
@@ -2,5 +2,6 @@ export enum DecisionType {
   CLAIMANT_IN_FAVOUR_OF_DEFENDANT = 'CLAIMANT_IN_FAVOUR_OF_DEFENDANT',
   CLAIMANT = 'CLAIMANT',
   DEFENDANT = 'DEFENDANT',
-  COURT = 'COURT'
+  COURT = 'COURT',
```
Comment (robertparkinson) [inline src/main/features/claimant-response/draft/courtDecision.ts]: another one of these... why do we have 3 of them?

File: src/main/features/claimant-response/views/confirmation/proposed-alternate-payment-plan-to-business-defendant.njk [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+<p>{{ t('The defendant should have sent you the details - send them to:') }}</p>
+
+<div class="keep-formatting panel">{{ t('Civil Money Claims') }}
+  {{ t('St. Katharine`s House') }}
+  {{ t('21-27 St. Katharine`s Street') }}
+  {{ t('Northampton') }}
+  {{ t('NN1 2LH') }}
+</div>
+
+<p>{{ t('Or email them to') }} <a href="mailto:moneyclaims@hmcts.gsi.gov.uk">moneyclaims@hmcts.gsi.gov.uk</a></p>
```
Comment (robertparkinson) [inline src/main/features/claimant-response/views/confirmation/proposed-alternate-payment-plan-to-business-defendant.njk]: this email should be retrieved from a config variable as it may change.

File: src/main/features/dashboard/views/macro/claimStatus/claimFullAdmissionForDefendant.njk (hunk 5 above)
Comment (robertparkinson) [reply src/main/features/dashboard/views/macro/claimStatus/claimFullAdmissionForDefendant.njk]: the only one of these three that is different is the by set date. please move the if in side there rather than duplicating the other content.

File: src/main/features/response/views/macro/confirmation/sendFinancialDetails.njk [hunk 15]
```diff
@@ -0,0 +1,7 @@
+{% from "internalLink.njk" import internalLink %}
+
+{% macro sendFinancialDetailsForBusiness(claimantName, claimantDetailsPageURI) %}
+  <h2 class="heading-medium">{{ t('Send your financial details') }}</h2>
+  <p>{{ t('Send {{ claimantName }} your company or organisation’s most recent statement of accounts, if you haven’t already.', { claimantName: claimantName }) }}</p>
+  <p>{{ internalLink('Get ' + claimantName + '’s contact details.', claimantDetailsPageURI) }}</p>
```
Comment (robertparkinson) [inline src/main/features/response/views/macro/confirmation/sendFinancialDetails.njk]: this needs to be translated

File: src/main/app/claims/models/claimant-response/court-determination/decisionType.ts (hunk 12 above)
Comment (rattur) [reply src/main/app/claims/models/claimant-response/court-determination/decisionType.ts]: Ok

File: src/main/features/response/views/macro/confirmation/sendFinancialDetails.njk (hunk 15 above)
Comment (rattur) [reply src/main/features/response/views/macro/confirmation/sendFinancialDetails.njk]: Yes I agree, will amend

File: src/main/features/claimant-response/draft/courtDecision.ts (hunk 13 above)
Comment (rattur) [reply src/main/features/claimant-response/draft/courtDecision.ts]: Yes we should have common area for Decision type, as for some reason we have duplicates within code base

File: src/integration-test/tests/citizen/claimantResponse/steps/claimant-reponse.ts [hunk 16]
```diff
@@ -99,14 +99,48 @@ export class ClaimantResponseSteps {
...
     taskListPage.selectTaskFreeMediation()
     freeMediationPage.accept()
     taskListPage.selectTaskCheckandSubmitYourResponse()
   }
 
+  acceptPartAdmitFromBusinessWithAlternativePaymentIntention (): void {
+    I.dontSee('COMPLETE')
+    taskListPage.selectTaskViewDefendantResponse()
+    defendantsResponsePage.submit()
+    defendantsResponsePage.submitHowTheyWantToPay() // bug
```
Comment (laurentaylor1) [inline src/integration-test/tests/citizen/claimantResponse/steps/claimant-reponse.ts]: `// bug`?

File: src/main/features/response/views/confirmation/part-admission.njk [hunk 17]
```diff
@@ -1,4 +1,7 @@
+{% from "../macro/confirmation/sendFinancialDetails.njk" import sendFinancialDetailsForBusiness %}
```
Comment (laurentaylor1) [inline src/main/features/response/views/confirmation/part-admission.njk]: Could this be made to not be a relative path? And all "s changed to 's
